rd-kt/rd-framework/src/main/kotlin/com/jetbrains/rd/framework/util/LifetimeCoroutineUtil.kt

```diff
@@ -108,4 +109,11 @@ fun Lifetime.createTerminatedAfter(duration: Duration, terminationContext: Corou
             delay(duration.toMillis())
             nested.terminate()
         }
-    }
\ No newline at end of file
+    }
+
+suspend fun <T> lifetimedCoroutineScope(lifetime: Lifetime, action: suspend CoroutineScope.() -> T) = coroutineScope {
```

Comment:
Some documentation wouldn't hurt here IMO. Something like "creates a coroutine scope that will be cancelled on lifetime termination, or parent scope cancellation".